axelrod/strategy_transformers.py

```diff
@@ -141,6 +153,27 @@ def __repr__(self):
                     prefix = ', '
                 return name
 
+            def reduce_for_decorated_class(self_):
```

Comment:
while checking PEP8 about `klass` , i noticed about the importance of sticking with `self` and `cls`.  even though it shadows a variable from outer scope, is it better to shadow the `self` variable than to use `self_` ?

here's the quote

> Function and method arguments
  Always use self for the first argument to instance methods.
  Always use cls for the first argument to class methods.